From 65fd68025eeb3c1b127e8cd86df7040877f7b7c4 Mon Sep 17 00:00:00 2001
From: Sebanc <sebanc.xda@gmail.com>
Date: Wed, 9 Oct 2019 21:49:47 +0200
Subject: [PATCH] Microsoft Surface devices: disable i915 Frame Buffer
 Compression

Disable Frame Buffer Compression as it appears to not work properly on Microsoft Surface devices.
---
 drivers/gpu/drm/i915/intel_fbc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_fbc.c b/drivers/gpu/drm/i915/intel_fbc.c
index 728a20e..6d96b3f 100644
--- a/drivers/gpu/drm/i915/intel_fbc.c
+++ b/drivers/gpu/drm/i915/intel_fbc.c
@@ -40,9 +40,14 @@
  * forcibly disable it to allow proper screen updates.
  */
 
+#include <linux/surface_devices_dmi.h>
+
 #include "intel_drv.h"
 #include "i915_drv.h"
 #include "intel_frontbuffer.h"
+
+static const struct dmi_system_id devices[] = surface_all_devices;
+
 static inline bool fbc_supported(struct drm_i915_private *dev_priv)
 {
 	return HAS_FBC(dev_priv);
@@ -789,6 +792,11 @@ static bool intel_fbc_can_enable(struct drm_i915_private *dev_priv)
 {
 	struct intel_fbc *fbc = &dev_priv->fbc;
 
+	if (dmi_check_system(devices)) {
+		fbc->no_fbc_reason = "unsupported on Microsoft Surface devices";
+		return false;
+	}
+
 	if (intel_vgpu_active(dev_priv)) {
 		fbc->no_fbc_reason = "VGPU is active";
 		return false;
-- 
2.20.1

